media-source/mediasource-activesourcebuffers.html

```diff
@@ -0,0 +1,212 @@
+                  {
+                      assert_equals(mediaSource.activeSourceBuffers.length, 1,
+                        "activeSourceBuffers updated when media element is loaded");
+                      assert_equals(mediaSource.activeSourceBuffers[0], sourceBuffer,
+                        "activeSourceBuffers contains sourceBuffer when media element is loaded");
+                      test.done();
+                  });
+              });
+          }, "SourceBuffer added to activeSourceBuffers list when its only video track gets loaded (and thus becomes selected).");
+
+
+          mediasource_test(function(test, mediaElement, mediaSource)
+          {
+              mediaElement.addEventListener("error", test.unreached_func("Unexpected event 'error'"));
+              MediaSourceUtil.fetchManifestAndData(test, manifestFilenameAudio, function (typeAudio, dataAudio)
+              {
+                  MediaSourceUtil.fetchManifestAndData(test, manifestFilenameVideo, function (typeVideo, dataVideo)
+                  {
+                      var sourceBufferAudio = mediaSource.addSourceBuffer(typeAudio);
+                      var sourceBufferVideo = mediaSource.addSourceBuffer(typeVideo);
+                      assert_equals(mediaSource.activeSourceBuffers.length, 0,
+                        "activeSourceBuffers is empty to start with");
+                      assert_equals(mediaSource.sourceBuffers.length, 2,
+                        "sourceBuffers list contains both SourceBuffers");
+                      assert_equals(mediaSource.sourceBuffers[0], sourceBufferAudio,
+                        "first SourceBuffer is the audio one");
+                      assert_equals(mediaSource.sourceBuffers[1], sourceBufferVideo,
+                        "second SourceBuffer is the video one");
+
+                      test.expectEvent(mediaElement, "loadedmetadata");
+                      test.expectEvent(mediaSource.activeSourceBuffers, "addsourcebuffer");
+                      sourceBufferAudio.appendBuffer(dataAudio);
+                      sourceBufferVideo.appendBuffer(dataVideo);
+
+                      test.waitForExpectedEvents(function()
+                      {
+                          assert_equals(mediaSource.activeSourceBuffers.length, 2,
+                            "activeSourceBuffers list updated when tracks are loaded");
+                          assert_equals(mediaSource.activeSourceBuffers[0], sourceBufferAudio,
+                            "first active SourceBuffer is the audio one");
+                          assert_equals(mediaSource.activeSourceBuffers[1], sourceBufferVideo,
+                            "second active SourceBuffer is the video one");
+                          test.done();
+                      });
+                  });
+              });
+          }, "Active SourceBuffers must appear in the same order as they appear in the sourceBuffers attribute.");
```

Comment:
A potentially better test would append the video first, since that might trigger out-of-order activeSourceBuffer sequence. A potentially best test would try both orderings. I'll see about doing the latter when merging this.